### PR TITLE
add kernel for squeeze_op, test=develop

### PR DESCRIPTION
**Pull request by JepsonWong**
Add kernel for squeeze_op.

- The SqueezeOp and SqueezeGradOp class originally inherited the framework::OperatorBase class, so it cannot be used in dygraph mode.

- By modifying the SqueezeOp and SqueezeGradOp class to inherit the framework::OperatorWithKernel class, this op can be used in the dygraph mode.

### CODE REVIEW

**Review by zhaoyuchen2018 — DISMISSED** [review]
LGTM

File: paddle/fluid/operators/squeeze_op.cc [hunk 1]
```diff
@@ -12,26 +12,32 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
...
 #include <string>
+#include <unordered_map>
 #include <vector>
 #include "paddle/fluid/framework/op_registry.h"
 
 namespace paddle {
 namespace operators {
 
-class SqueezeOpInferShape : public framework::InferShapeBase {
+// TODO(paddle-dev): Should use OpKernel.
```
Comment (JiabinYang) [inline paddle/fluid/operators/squeeze_op.cc]: why add this TODO?

**Review by JiabinYang — COMMENTED** [review]
SomeComments

File: paddle/fluid/operators/squeeze_op.cc (hunk 1 above)
Comment (JepsonWong) [reply paddle/fluid/operators/squeeze_op.cc]: done

**Review by JiabinYang — APPROVED** [review]
LGTM

**Pull request merged**